ext/crates/fp/src/field/mod.rs

```diff
@@ -138,16 +143,42 @@ mod tests {
         assert_eq!(elements, expansions);
     }
 
-    /// Given a function that generates elements of a field, test the field axioms and good behavior
-    /// of the Frobenius endomorphism. The function should have signature
-    ///
-    /// ```ignore
-    /// fn arb_elements<const N: usize>() -> impl Strategy<Value = (F, [FieldElement<F>; N])>
-    /// ```
-    #[macro_export]
+    pub(super) fn arb_elements<F: Field, const N: usize>(
+    ) -> impl Strategy<Value = (F, [FieldElement<F>; N])> {
+        any::<F>().prop_flat_map(move |f| {
+            let elements: [_; N] = (0..N)
+                .map(|_| f.arb_element())
+                .collect::<Vec<_>>()
+                .try_into()
+                .unwrap();
+            (Just(f), elements)
+        })
```

Comment:
I just found out about `std::array:from_fn`. This can just be
```rust
any::<F>().prop_flat_map(move |f| (Just(f), std::array::from_fn(|_| f.arb_element())))
```
but I'll make that change after the merge so it doesn't disrupt this PR and #169 too much